File: python/agent.py
from base import BaseAgent, TurnData, Action


class Agent(BaseAgent):

    def do_turn(self, turn_data: TurnData) -> Action:
        print(f"TURN {self.max_turns - turn_data.turns_left}/{self.max_turns}")
        for agent in turn_data.agent_data:
            print(f"AGENT {agent.name}")
            print(f"POSITION: {agent.position}")
            print(f"CARRYING: {agent.carrying}")
            print(f"COLLECTED: {agent.collected}")
        for row in turn_data.map:
            print(''.join(row))
        return Action.DOWN


if __name__ == '__main__':
    winner = Agent().play()
    print("WINNER: " + winner)

File: java/src/Agent.java
import base.Action;
import base.AgentData;
import base.BaseAgent;
import base.TurnData;

import java.io.IOException;
import java.util.Arrays;

public class Agent extends BaseAgent {

    public Agent() throws IOException {
        super();
    }

    @Override
    public Action doTurn(TurnData turnData) {
        System.out.println("TURN " + (maxTurns - turnData.turnsLeft) + "/" + maxTurns);
        for (AgentData agent : turnData.agentData) {
            System.out.println("AGENT " + agent.name);
            System.out.println("POSITION: (" + agent.position.row + ", " + agent.position.column + ")");
            System.out.println("CARRYING: " + agent.carrying);
            System.out.println("COLLECTED: " + Arrays.toString(agent.collected));
        }
        for (int i = 0; i < gridSize; i++) {
            for (int j = 0; j < gridSize; j++)
                System.out.print(turnData.map[i][j]);
            System.out.println();
        }
        return Action.UP;
    }

    public static void main(String[] args) {
        try {
            String winner = new Agent().play();
            System.out.println("WINNER: " + winner);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}
